### PR TITLE
Patch to prevent logging endchat exception when conversation was disconnected

### PR DESCRIPTION
**Pull request by elopezanaya**
### Id of the task, bug, story or other reference.

4345582

### Description
_Include a description of the problem to be solved_

When c2 close the widget for a session that is already ended by agent, backend throws an exception which doesnt reflect the result, since the conversation was already closed by agent.

We still want to call the endchat in case close status has not been propagated in DV, causing the conversation to stays open.

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

Evaluate state to check disconnection was received, as cause of chat ended by agent, or iddle activity.

We still want to call the endchat in case close status has not been propagated in DV, causing the conversation to stays open.

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

**Review by mithunjohny — APPROVED** [review]

**Review by matan-hershberg-nuance — APPROVED** [review]

File: chat-widget/src/components/livechatwidget/common/endChat.ts [hunk 1]
```diff
@@ -131,12 +131,18 @@ const endChat = async (props: ILiveChatWidgetProps, chatSDK: any, state: ILiveCh
...
+            const inMemoryState = executeReducer(state, { type: LiveChatWidgetActionType.GET_IN_MEMORY_STATE, payload: null });
+            // if the chat was disconnected or ended by the agent, we don't want to log the error
+            if (!inMemoryState?.appStates?.chatDisconnectEventReceived) {
+                TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
+                    Event: TelemetryEvent.EndChatSDKCallFailed,
+                    ExceptionDetails: {
+                        exception: ex
+                    }
+                });
+            }
```
Comment (dleblondMSFT) [inline chat-widget/src/components/livechatwidget/common/endChat.ts:144]: Should we log a warning in the else branch here for DRIs?

**Review by matan-hershberg-nuance — APPROVED** [review]

**Review by dleblondMSFT — APPROVED** [review]
